src/kotlin/kage/crypto/chacha20/ChaCha20Poly1305OutputStream.kt

```diff
@@ -1,81 +0,0 @@
-/**
- * Copyright 2021 The kage Authors. All rights reserved. Use of this source code is governed by
- * either an Apache 2.0 or MIT license at your discretion, that can be found in the LICENSE-APACHE
- * or LICENSE-MIT files respectively.
- */
-package kage.crypto.chacha20
-
-import java.io.OutputStream
-import kage.crypto.chacha20.ChaCha20Poly1305.CHACHA_20_POLY_1305_NONCE_LENGTH
-
-// TODO: This is not thread safe
-internal class ChaCha20Poly1305OutputStream(
-  private val key: ByteArray,
-  private val dst: OutputStream
-) : OutputStream() {
-
-  private val nonce = ByteArray(CHACHA_20_POLY_1305_NONCE_LENGTH)
-
-  private val buf = ByteArray(CHUNK_SIZE)
-  private var bufSize = 0
-
-  override fun write(i: Int) {
-    val b = i.toByte()
-
-    buf[bufSize] = b
-    bufSize++
-
-    if (bufSize == CHUNK_SIZE) {
-      flushChunk()
-      bufSize = 0
-    }
-  }
-
-  override fun close() {
-    flushChunk(last = true)
-    // Not closing underlying stream
-  }
-
-  /*
```

Comment:
I was wrong here and fixed this now.

Age v1.0.0 does not implement this from the spec, but the age version in main does. So I changed this to comply to the spec.

The solution was not so complicated as I thought, we just need to hold off on flushing until the stream is closed, this way when we close the stream we can set the last chunk flag and everything is according to spec.